diff --git a/node_modules/react-native-cloud-store/ios/CloudStore.swift b/node_modules/react-native-cloud-store/ios/CloudStore.swift
index eb2b3e2..da7a403 100644
--- a/node_modules/react-native-cloud-store/ios/CloudStore.swift
+++ b/node_modules/react-native-cloud-store/ios/CloudStore.swift
@@ -410,7 +410,10 @@ extension CloudStoreModule {
                     .contentModificationDateKey,
                 .creationDateKey,
                 .nameKey,
-                .localizedNameKey
+                .localizedNameKey,
+                .fileSizeKey,
+                .isDirectoryKey,
+                .ubiquitousItemIsExcludedFromSyncKey
             ])
             let dict = NSMutableDictionary()
 
@@ -440,6 +443,9 @@ extension CloudStoreModule {
             }
             dict["name"] = resources.name
             dict["localizedName"] = resources.localizedName
+            dict["fileSize"] = resources.fileSize
+            dict["isDirectory"] = resources.isDirectory
+            dict["ubiquitousItemIsExcludedFromSync"] = resources.ubiquitousItemIsExcludedFromSync
 
             resolve(dict)
         } catch {
