diff --git a/node_modules/react-native-hold-menu/src/components/backdrop/Backdrop.tsx b/node_modules/react-native-hold-menu/src/components/backdrop/Backdrop.tsx
index dc5fa35..13bc123 100644
--- a/node_modules/react-native-hold-menu/src/components/backdrop/Backdrop.tsx
+++ b/node_modules/react-native-hold-menu/src/components/backdrop/Backdrop.tsx
@@ -13,7 +13,7 @@ import {
 } from 'react-native-gesture-handler';
 
 // Components
-import { BlurView } from 'expo-blur';
+import { VibrancyView as BlurView} from '@react-native-community/blur';
 
 // Utils
 import { styles } from './styles';
@@ -94,7 +94,7 @@ const BackdropComponent = () => {
 
   const animatedContainerProps = useAnimatedProps(() => {
     return {
-      intensity: withTiming(
+      blurAmount: withTiming(
         state.value === CONTEXT_MENU_STATE.ACTIVE ? 100 : 0,
         {
           duration: HOLD_ITEM_TRANSFORM_DURATION,
@@ -115,8 +115,7 @@ const BackdropComponent = () => {
   return (
     <TapGestureHandler onHandlerStateChange={tapGestureEvent}>
       <AnimatedBlurView
-        // @ts-ignore
-        tint="default"
+        blurType={theme.value === 'light' ? 'materialLight' : 'materialDark'}
         animatedProps={animatedContainerProps}
         style={[styles.container, animatedContainerStyle]}
       >
diff --git a/node_modules/react-native-hold-menu/src/components/menu/MenuList.tsx b/node_modules/react-native-hold-menu/src/components/menu/MenuList.tsx
index 875d6b5..9cb7b13 100644
--- a/node_modules/react-native-hold-menu/src/components/menu/MenuList.tsx
+++ b/node_modules/react-native-hold-menu/src/components/menu/MenuList.tsx
@@ -16,7 +16,7 @@ import {
   calculateMenuHeight,
   menuAnimationAnchor,
 } from '../../utils/calculations';
-import { BlurView } from 'expo-blur';
+import { VibrancyView as BlurView} from '@react-native-community/blur';
 
 import MenuItems from './MenuItems';
 
@@ -107,7 +107,7 @@ const MenuListComponent = () => {
   }, [theme]);
 
   const animatedProps = useAnimatedProps(() => {
-    return { tint: theme.value };
+    return { blurType: theme.value === 'dark' ? 'materialDark' : 'materialLight' };
   }, [theme]);
 
   const setter = (items: MenuItemProps[]) => {
@@ -127,7 +127,7 @@ const MenuListComponent = () => {
 
   return (
     <AnimatedView
-      intensity={100}
+      blurAmount={100}
       animatedProps={animatedProps}
       style={[styles.menuContainer, messageStyles]}
     >
